uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/leave-request-api_test.js

```diff
@@ -115,5 +147,259 @@ define([
+        promise = LeaveRequestAPI.calculateBalanceChange(requestData);
+      });
+
+      afterEach(function () {
+        $httpBackend.flush();
+      });
+
+      it('calls endpoint', function () {
+        promise.then(function (result) {
+          expect(LeaveRequestAPI.calculateBalanceChange).toHaveBeenCalled();
+        });
+      });
+
+      it('returns expected data keys', function () {
+        promise.then(function (result) {
+          //returns an object(associative array) and not an array
+          expect(result.amount).toBeDefined();
+          expect(result.breakdown).toBeDefined();
+
+          var breakdown = result.breakdown[0];
+          expect(breakdown.date).toBeDefined();
+          expect(breakdown.amount).toBeDefined();
+          expect(breakdown.type).toBeDefined();
+
+          var breakdownType = breakdown.type;
+          expect(breakdownType.id).toBeDefined();
+          expect(breakdownType.value).toBeDefined();
+          expect(breakdownType.label).toBeDefined();
+        });
+      });
+
+      it('returns expected data values', function () {
+        promise.then(function (result) {
+          expect(result.amount).toEqual(jasmine.any(Number));
+          expect(result.breakdown).toEqual(jasmine.any(Object));
+
+          var breakdown = result.breakdown[0];
+          expect(breakdown.amount).toEqual(jasmine.any(Number));
+          expect(moment(breakdown.date, dateFormat, true).isValid()).toBe(true);
+          expect(breakdown.type).toEqual(jasmine.any(Object));
+
+          var breakdownType = breakdown.type;
+          expect(breakdownType.id).toEqual(jasmine.any(Number));
+          expect(breakdownType.value).toEqual(jasmine.any(Number));
+          expect(mockData.getAllAbsenceTypesTitles()).toContain(breakdownType.label);
+        });
+      });
+
+      describe('error handling', function () {
+
+        beforeEach(function () {
+          errorObject = {
+            is_error: 1,
+            error_message: 'contact_id, from_date and from_date_type in params are mandatory'
+          };
+          requestData = {};
+          promise = LeaveRequestAPI.calculateBalanceChange(requestData);
+        });
+
+        afterEach(function () {
+          $rootScope.$apply();
+        });
+
+        it('throws an error', function () {
+          promise.catch(function (result) {
+            expect(result).toEqual(errorObject);
+          });
+        });
+      });
+    });
+
+    describe('create()', function () {
+
+      beforeEach(function () {
+        requestData = mockData.createRandomLeaveRequest();
+        spyOn(LeaveRequestAPI, 'create').and.callThrough();
+        promise = LeaveRequestAPI.create(requestData);
+      });
+
+      afterEach(function () {
+        $httpBackend.flush();
+      });
+
+      it('call endpoint', function () {
+        promise.then(function (result) {
+          expect(LeaveRequestAPI.create).toHaveBeenCalled();
```

Comment:
Again, same thing. You just manually called the method *in the test*, what's the point of testing if it has indeed been called?